src/macros/macrorecorder.cpp

```diff
@@ -0,0 +1,106 @@
+#include "macrorecorder.h"
+
+#include <QtConcurrentRun>
+
+#include "control/controlproxy.h"
+#include "preferences/configobject.h"
+
+namespace {
+constexpr uint kMaxMacroSize = 1000;
+const QString kConfigGroup = QStringLiteral("[MacroRecording]");
+}
+
+MacroRecorder::MacroRecorder()
+        : m_COToggleRecording(ConfigKey(kConfigGroup, "recording_toggle")),
+          m_CORecStatus(ConfigKey(kConfigGroup, "recording_status")),
```

Comment:
Can this become just "status"?

Reply:
I think this should become read only by m_CORecStatus->setReadOnly();
You need to use forceSet() to set it anyway.

Reply:
Peviously we considered to have more "toplevel"-controls, but yeah probably
and yes read only is correct

Reply:
Why would I need forceSet anyway?